haskell/test/SMT/Common/Gen.hs

```diff
@@ -0,0 +1,426 @@
+-- SPDX-FileCopyrightText: 2021 TQ Tezos
+-- SPDX-License-Identifier: LicenseRef-MIT-TQ
+
+-- | Generate the storage and entrypoint call
+module SMT.Common.Gen
+  ( genMkModelInput
+  , genXtzTransferType
+  , genTokenTransferType
+  ) where
+
+import Universum hiding (drop, swap)
+
+import Crypto.Random (drgNewSeed, seedFromInteger, withDRG)
+import qualified Data.Map as Map
+import Hedgehog.Gen.Tezos.Address (genAddress)
+import qualified Hedgehog.Gen as Gen
+import qualified Hedgehog.Range as Range
+
+import Lorentz hiding (not, cast, concat, get)
+import qualified Lorentz.Contracts.Spec.FA2Interface as FA2
+import Michelson.Test.Dummy
+import Hedgehog.Gen.Tezos.Crypto (genSecretKey)
+import Tezos.Address (Address(..), unsafeParseContractHash, mkKeyAddress)
+import Tezos.Crypto (SecretKey, toPublic, sign)
+import Util.Named ((.!))
+
+import Ligo.BaseDAO.Common.Types
+import Ligo.BaseDAO.Types
+import SMT.Model.BaseDAO.Types
+import SMT.Common.Types
+
+genMkModelInput :: SmtOption -> GeneratorT MkModelInput
+genMkModelInput option@SmtOption{..} = do
+
+  -- Initial
+  addrs <- genSecretKey
+    <&> (\secret -> (mkKeyAddress . toPublic $ secret, secret))
+    & vectorOf poolSize
+  startLevel <- Gen.integral (Range.constant 0 10)
+
+  modify (\s -> s { gsAddresses = addrs, gsLevel = startLevel, gsMkGenPropose = soMkPropose, gsMkCustomCalls = soMkCustomCalls })
+
+  -- Run Gen
+  mkContractCalls <-
+    [1..groupCallAmount]
+       &  mapM (\(_ :: Integer) -> genContractCallGroup)
+
+  mkModelState <- genModelState option
+
+  pure $ \args@ModelInputArg{..} -> ModelInput
+    ( mkContractCalls <&> (\f -> f args)
+                      & concat
+    , mkModelState miaGuardianAddr miaGovAddr)
+  where
+    groupCallAmount = 2
```

Comment:
It will make 2 contract call groups. I generate contract calls in group since, it make the calls more likely to be succeed (Some needs to be called in order). One of the group is generate random calls, that covers the random call case as well.

I've used:
(SMT.RegistryDAO.hs)
```
    withTests 30 $ property $ do
      runBaseDaoSMT option
```

So each SMT test will run 30 times, I could make that bigger if necessary.